control-plane/agents/core/src/core/scheduling/mod.rs

```diff
@@ -72,6 +75,42 @@ impl PoolFilters {
     pub(crate) fn usable(_: &GetSuitablePoolsContext, item: &PoolItem) -> bool {
         item.pool.status != PoolStatus::Faulted && item.pool.status != PoolStatus::Unknown
     }
+    /// Should only attempt to use pools having msp-operator creation label iff topology has it
+    pub(crate) fn topology(request: &GetSuitablePoolsContext, item: &PoolItem) -> bool {
+        let creation_label: String = format!("{}:{}", OPENEBS_CREATED_BY_KEY, MSP_OPERATOR);
+        let pool_has_creation_label = match request.registry().specs().get_pool(&item.pool.id) {
+            Ok(spec) => match spec.labels {
+                None => false,
+                Some(label) => {
+                    label.contains_key(&*String::from(OPENEBS_CREATED_BY_KEY))
+                        && label.get(&*String::from(OPENEBS_CREATED_BY_KEY))
+                            == Some(&String::from(MSP_OPERATOR))
+                }
+            },
+            Err(_) => false,
```

Comment:
If we get an error, it is because we failed to get the pool. Can't we just return false here rather than setting `pool_has_creation_label` to false and continuing?

Reply:
Yes. We should end here.